wrappers/kn/README.md

```diff
@@ -0,0 +1,40 @@
+# ZXing-C++ Kotlin/Native Library
+
+## Install
+
+The easiest way to use the library is to fetch if from _mavenCentral_. Simply add
+
+```gradle
+implementation("io.github.zxing-cpp:kotlin-native:2.2.1")
+```
+
+to your `build.gradle.kts` file in the `dependencies` section of `nativeMain` source set.
+
+## Use
+
+A trivial use case looks like this:
+
+```kotlin
+import zxingcpp.BarcodeReader
+import zxingcpp.ImageView
+
+val options = ReaderOptions()
+val barcodeReader = BarcodeReader(options)
```

Comment:
Actually, it funny, kotlin have "global static function (outside of a class)", although most of us regard kotlin as a object oriented lang. They are usually used for writing helper functions for now. (Sorry for off topic)

Reply:
Done in: https://github.com/zxing-cpp/zxing-cpp/pull/719/commits/3480467435703c2c062749b0c5bf691e3c2c3f00

Reply:
In android development or sth else, `ReaderOptions` is seprated from `Reader` object, because `ReaderOption` might be injected from some place. And the Reader object is not good for passing everywhere, because they usually contains reference to a `context`. If you pass them everywhere, there might be memory leaks. But `ReaderOption` is usually a pure value class, and don't suffer from that.
But considering our `BarcodeReader` is not mapped to a real c object(and I think the probability of that happening in the future is also very low).


That's what I think.

Reply:
Thanks but I can't quite follow your thoughts here: what `context` are you referring to? There must not be memory leaks and in a GC environment, technically there can't be any? `ReaderOptions` in this context (k/n) are not a value class and technically there is no difference between passing the handle to the struct via parameter of a (static) function or via the implicit `this` pointer in a class context. The only difference is the usability/ergonomics of it and in that regard I am convinced that having two classes instead of one is adding complexity without adding value.

But I see your point that having the class named `Options` would make people more comfortable using it as a vocabulary type and pass it around their code, even though they could do the same thing just as well will a class called `Reader`.